provider/pkg/gen/schema.go

```diff
@@ -1008,6 +1014,43 @@ func (g *packageGenerator) setOperationsBaseURL(cloudOp *resources.CloudAPIOpera
 	cloudOp.Operations.EmbeddedOperationField = embeddedOperationFieldName
 }
 
+func (g *packageGenerator) dedupeId(typeName string, resourcePropBag *propertyBag) {
+	newId := apiPropNameToSdkName(typeName, typeName+"Id")
+
+	// First check if an existing property with the name id exists in the resource properties (at the top level).
+	// Replace with the newId if so.
+	props, exists := resourcePropBag.specs["id"]
+	if !exists {
+		return
+	}
+
+	tok := g.genToken(typeName)
+	ignoreCollision := false
+	if _, exists = resourcePropBag.specs[newId]; exists {
+		if tok == "google-native:baremetalsolution/v2:NfsShare" {
+			ignoreCollision = true
```

Comment:
Might be worth a comment explaining why we want to ignore collisions here.

Reply:
Done